src/App.tsx

```diff
@@ -1,15 +1,82 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+function filterPerson(arr: Person[], query: string) {
+  return query ? [...arr].filter(elem => elem.name.includes(query)) : [...arr];
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [value, setValue] = useState('');
+  const [appliedValue, setAppliedValue] = useState('');
+  const [title, setTitle] = useState('No selected person');
+
+  const [showArray, setShowArray] = useState(false);
+  const timerId = useRef(0);
+  const showMessageRef = useRef(false);
+
+  const updatedPersons = useMemo(() => {
+    return filterPerson(peopleFromServer, appliedValue);
+  }, [appliedValue]);
+
+  const handleChangeInputName = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setValue(event.target.value);
+      setTitle('No selected person');
+      window.clearTimeout(timerId.current);
+
+      timerId.current = window.setTimeout(() => {
+        setAppliedValue(!!event.target.value ? event.target.value : '');
+      }, 300);
+    },
+    [],
```

Comment:
The `useCallback` hook is used without any dependencies. This means that the function will be recreated on every render, which might not be optimal. Consider adding dependencies to this hook if there are any variables that the function relies on.